tests/test_restore_command.rs

```diff
@@ -100,3 +102,124 @@ fn test_restore() {
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    conflict
+    ├─╮
+    o │  b
+    │ o  a
+    ├─╯
+    o  base
+    o
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file")).unwrap()
+        , @r###"
+            <<<<<<<
+            %%%%%%%
+            -base
+            +a
+            +++++++
+            b
+            >>>>>>>
+            "###);
+    std::fs::write(repo_path.join("file"), "resolution").unwrap();
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff"]), 
+    @r###"
+    Resolved conflict in file:
+       1     : <<<<<<<
+       2     : %%%%%%%
+       3     : -base
+       4     : +a
+       5     : +++++++
+       6     : b
+       7     : >>>>>>>
+            1: resolution
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["restore", "file"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Created 63198ca2e4aa conflict
+    Working copy now at: 63198ca2e4aa conflict
+    Added 0 files, modified 1 files, removed 0 files
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -base
+    +a
+    +++++++
+    b
+    >>>>>>>
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diff"]);
+    insta::assert_snapshot!(stdout, @"");
+    // The same, but without the `file` argument
+    std::fs::write(repo_path.join("file"), "resolution").unwrap();
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff"]), 
+    @r###"
+    Resolved conflict in file:
+       1     : <<<<<<<
+       2     : %%%%%%%
+       3     : -base
+       4     : +a
+       5     : +++++++
+       6     : b
+       7     : >>>>>>>
+            1: resolution
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["restore"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Created d955febceac1 conflict
+    Working copy now at: d955febceac1 conflict
+    Added 0 files, modified 1 files, removed 0 files
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -base
+    +a
+    +++++++
+    b
+    >>>>>>>
+    "###);
+}
+
+fn create_commit(
+    test_env: &TestEnvironment,
+    repo_path: &Path,
+    name: &str,
+    parents: &[&str],
+    files: &[(&str, &str)],
+) {
+    if parents.is_empty() {
+        test_env.jj_cmd_success(repo_path, &["new", "root", "-m", name]);
+    } else {
+        let mut args = vec!["new", "-m", name];
+        args.extend(parents);
+        test_env.jj_cmd_success(repo_path, &args);
+    }
+    for (name, content) in files {
+        std::fs::write(repo_path.join(name), content).unwrap();
+    }
+    test_env.jj_cmd_success(repo_path, &["branch", "create", name]);
+}
```

Comment:
This looks familiar :) We should put it in `tests/common/mod.rs` (or similar), but that can be separate from this PR.

Reply:
Yeah. I also use a slightly different version of it in https://github.com/martinvonz/jj/pull/1146. I'm not sure if there's a way to have one version that works everywhere.

In any case, I won't do it here.

As usual, thanks for the review!